test/specs/record_type_test.dart

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:code_builder/code_builder.dart';
+import 'package:code_builder/src/specs/type_record.dart';
+import 'package:test/test.dart';
+
+import '../common.dart';
+
+void main() {
+  late DartEmitter emitter;
+
+  useDartfmt();
+
+  setUp(() => emitter = DartEmitter.scoped(useNullSafetySyntax: true));
+
+  final intRef = TypeReference((b) => b.symbol = 'int');
+  TypeReference listRef(TypeReference argType) => TypeReference((b) => b
+    ..symbol = 'List'
+    ..types.add(argType));
+
+  test('should create a record type with positional fields', () {
+    expect(
+      RecordType((b) => b
+        ..positionalFieldTypes.addAll(
+            [intRef, listRef(intRef).rebuild((b) => b..isNullable = true)])
+        ..isNullable = true),
+      equalsDart('(int, List<int>?)?', emitter),
+    );
+  });
+
+  test('should create a record type with one positional field', () {
+    expect(RecordType((b) => b..positionalFieldTypes.add(intRef)),
+        equalsDart('(int,)', emitter));
```

Comment:
These are great, can we add one for the empty record type (`()`) (Just looked it up, very strange but it's legal)